src/js/views/maps/ToolbarView.js

```diff
@@ -183,6 +193,19 @@ define(
           //   view: DrawTool,
           //   viewOptions: {}
           // },
+          {
+            label: 'Viewfinder',
+            icon: 'search',
+            view: ViewfinderView,
+            action(view, model) {
+              const sectionEl = this;
+              view.defaultActivationAction(sectionEl);
+              sectionEl.sectionView.focusInput();
```

Comment:
I think it would be more typical (I have little experience with Backbone, so I'm speaking from general UI framework experience) for the view to be rendered when it was activated/shown.  That way we don't run code that we don't need to. If it was expensive to load one of these views, for example, we wouldn't want to load it unless the user requested it. 

I could see there being an option for actionWithDefault or something to that extent. Conversely, we could add logic that says "if there is an action and a view defined, then run the default action then call the custom action". I think that's the most developer friendly solution

Reply:
Thanks, that's super useful. Made an issue to address these changes here: https://github.com/NCEAS/metacatui/issues/2269